src/legacy/core_plugins/kbn_vislib_vis_types/public/components/options/heatmap/index.tsx

```diff
@@ -0,0 +1,190 @@
+            [paramName]: value,
+          },
+        },
+      ]),
+    [valueAxis, setValue]
+  );
+
+  useEffect(() => {
+    setValidity(stateParams.setColorRange ? isColorRangesValid : !isColorsNumberInvalid);
+  }, [stateParams.setColorRange, isColorRangesValid, isColorsNumberInvalid, setValidity]);
+
+  return (
+    <>
+      <EuiPanel paddingSize="s">
+        <EuiTitle size="xs">
+          <h2>
+            <FormattedMessage
+              id="kbnVislibVisTypes.editors.heatmap.basicSettingsTitle"
+              defaultMessage="Basic settings"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+
+        <BasicOptions {...props} />
+
+        <SwitchOption
+          label={i18n.translate('kbnVislibVisTypes.editors.heatmap.highlightLabel', {
+            defaultMessage: 'Highlight range',
+          })}
+          paramName="enableHover"
+          value={stateParams.enableHover}
+          setValue={setValue}
+          tooltip={i18n.translate('kbnVislibVisTypes.editors.heatmap.highlightLabelTooltip', {
+            defaultMessage:
+              'Highlight hovered range in the chart and corresponding label in the legend.',
+          })}
+        />
+      </EuiPanel>
+
+      <EuiSpacer size="s" />
+
+      <EuiPanel paddingSize="s">
+        <EuiTitle size="xs">
+          <h2>
+            <FormattedMessage
+              id="kbnVislibVisTypes.editors.heatmap.heatmapSettingsTitle"
+              defaultMessage="Heatmap settings"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+
+        <ColorSchemaOptions
+          colorSchema={stateParams.colorSchema}
+          colorSchemas={vis.type.editorConfig.collections.colorSchemas}
+          invertColors={stateParams.invertColors}
+          uiState={uiState}
+          setValue={setValue as SetColorSchemaOptionsValue}
+        />
+        <EuiSpacer size="s" />
+
+        <SelectOption
+          label={i18n.translate('kbnVislibVisTypes.controls.heatmapOptions.colorScaleLabel', {
+            defaultMessage: 'Color scale',
+          })}
+          options={vis.type.editorConfig.collections.scales}
+          paramName="type"
+          value={valueAxis.scale.type}
+          setValue={setValueAxisScale}
+        />
+
+        <SwitchOption
+          label={i18n.translate(
+            'kbnVislibVisTypes.controls.heatmapOptions.scaleToDataBoundsLabel',
+            { defaultMessage: 'Scale to data bounds' }
+          )}
+          paramName="defaultYExtents"
+          value={valueAxis.scale.defaultYExtents}
+          setValue={setValueAxisScale}
+        />
+
+        <SwitchOption
+          disabled={stateParams.setColorRange}
+          label={i18n.translate('kbnVislibVisTypes.controls.heatmapOptions.percentageModeLabel', {
+            defaultMessage: 'Percentage mode',
+          })}
+          paramName="percentageMode"
+          value={stateParams.setColorRange ? false : stateParams.percentageMode}
```

Comment:
nit: Why hiding the value of this switch if it's disabled? This isn't done in other places (e.g. label settings):
<img width="345" alt="Screenshot 2019-09-25 at 18 13 20" src="https://user-images.githubusercontent.com/1508364/65619297-30fb4800-dfc0-11e9-89e3-af6f8938c9da.png">

Reply:
Because it can be confusing if the user sees "4 colors" in the config panel but that's really being overridden by the custom ranges they've established which may be 3 colors.

Reply:
@cchaos 
The `Percentage mode` switch (which Joe asked about) does not have any relation to the `Number of colors`, it is a standalone param which is excluded from visualization config if `Use custom ranges` is turned on! The same work for `Number of colors`.
I suppose @flash1293 asked why we didn't do the same in the `Labels` panel -> means to `turn off` all of params (`Rotate` and `Overwrite automatic color`) if `Show labels` is off.
Could you please confirm it is a right UX solution ? Thanks

Reply:
I think I'm still confused, can you show me some screenshots?